Sources/System/FilePath/FilePathSyntax.swift

```diff
@@ -0,0 +1,353 @@
+// Query API
+// @available(...)
+extension FilePath {
+  /// Returns true if this path uniquely identifies the location of
+  /// a file without reference to an additional starting location.
+  ///
+  /// On Unix platforms, absolute paths begin with a `/`. `isAbsolute` is equivalent
+  /// to `root != nil`.
+  ///
+  /// On Windows, absolute paths are fully qualified paths. UNC paths and device paths
+  /// are always absolute. Traditional DOS paths are absolute if they begin with a volume or drive
+  /// followed by a `:` and a separator.
+  ///
+  /// NOTE: This does not perform shell expansion or substitute
+  /// environment variables; paths beginning with `~` are considered relative.
+  ///
+  /// Examples:
+  /// * Unix:
+  ///   * `/usr/local/bin`
+  ///   * `/tmp/foo.txt`
+  ///   * `/`
+  /// * Windows:
+  ///   * `C:\Users\`
+  ///   * `\\?\UNC\server\share\bar.exe`
+  ///   * `\\server\share\bar.exe`
+  ///
+  /// TODO: We should add API to system to access environment variables
+  /// and perform expansion/substituion, but that's deferred for now. Should
+  /// we mention that?
+  public var isAbsolute: Bool {
+    self.root?._rootIsAbsolute ?? false
+  }
+
+  /// Returns true if this path is not absolute (see absolute)
+  ///
+  /// Examples:
+  /// * Unix:
+  ///   * `~/bar`
+  ///   * `tmp/foo.txt`
+  /// * Windows:
+  ///   * `bar\baz`
+  ///   * `C:Users\`
+  ///   * `\Users`
+  public var isRelative: Bool { !isAbsolute }
+
+  /// A textual representation of the file path using the generic separator (`/`).
+  ///
+  /// On Unix, this is equivalent to `description`
+  ///
+  /// TODO: Should `dump()` use this? Also, should dump escape invalid Unicode?
+  public var genericDescription: String {
+    fatalError()
+  }
+
+  /// Returns whether `self` is a prefix of `other`
+  public func starts(with other: FilePath) -> Bool {
+    fatalError()
+  }
+
+  /// Returns whether `self` is a suffix of `other`
+  public func ends(with other: FilePath) -> Bool {
```

Comment:
I might be reading this wrong, but is the description backwards?  If self ends with other, then other is a suffix of self.

Reply:
Yup, same